en/08_Changelogs/6.0.0.md

```diff
@@ -928,6 +929,33 @@ The following classes have been changed when code was moved to the `silverstripe
 |`SilverStripe\UserForms\Form\UserFormsRequiredFields`|[`SilverStripe\UserForms\Form\UserFormsRequiredFieldsValidator`](api:SilverStripe\UserForms\Form\UserFormsRequiredFieldsValidator)|
 |`Symbiote\AdvancedWorkflow\Forms\AWRequiredFields`|[`Symbiote\AdvancedWorkflow\Forms\AWRequiredFieldsValidator`](api:Symbiote\AdvancedWorkflow\Forms\AWRequiredFieldsValidator)|
 
+### DataObject `validate()` signature changed {#dataobject-validate}
+
+The method signature of the `validate()` method on `DataObject` has changed as it is now strongly typed. It now explicitly returns a [`ValidationResult`](api:SilverStripe\Core\Validation\ValidationResult). Previously the method still returned a `ValidationResult`, though this was not defined in the method signature.
+
+What this means is that you will now need to update the method sigature of any custom `validate()` methods you have in your `DataObject` subclasses to match the new signature.
+
+```php
+// app/src/Models/MyDataObject.php
+namespace App\Models;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\ORM\DataObject;
+
+class MyDataObject extends DataObject
+{
+    // ...
+    public function validate(): ValidationResult
+    {
+        $result = parent::validate();
+        if ($this->Title === 'Bad Title') {
+            $result->addError('Title cannot be "Bad Title"');
+        }
+        return $result;
+    }
+}
+```
```

Comment:
I don't think this is worth its own section. There's a _lot_ of signature changes happening.
It might make sense for this to be a sub-section of "Validation added to DBFields" though, since this is a result of that work.

Alternatively "Validation added to DBFields" could have a parent "Changes to validation" heading, and we could move "Changes to password validation" and any other validation-related changelog entries in there.

Reply:
DataObject validate should not be a subsection of DBField validation IMO. Also the DBField work is new functionality, this is just a signature change on something presumably fairly high use.

I think I'll just delete this and leave it in as a single line entry. PHP will tell people quicky enough how to update their method signatures.